src/interface/swarm.cpp

```diff
@@ -853,8 +851,8 @@ void Swarm::LoadBuffers_(const int max_indices_size) {
   auto particle_indices_to_send = particle_indices_to_send_;
   pmb->par_for(
       "Pack Buffers", 0, max_indices_size,
-      KOKKOS_LAMBDA(const int n) {        // Max index
-        for (int m = 0; m < nbmax; m++) { // Number of neighbors
+      KOKKOS_LAMBDA(const int n) {                        // Max index
+        for (int m = 0; m < pmb->pbval->nneighbor; m++) { // Number of neighbors
```

Comment:
I'm not sure `pmb->pbval->nneighbor` woud be available on a device (I'd actually be surprised if it'd would be).
So this could be another issue that still causes problems.

Reply:
Ah good point, my mistake. Let's see if the Darwin CI passes now...

Reply:
It looks like this definitely fixed *something*. Now only the leapfrog outflow fails, as before (instead of all 4 leapfrog tests). So we should probably focus on what's different with the outflow test compared to the regular leapfrog test. Although, @brryan did you say you also saw segfaults in the regular leapfrog test if you ran them for longer?

Reply:
Yeah looks like that only fixed the bug I introduced earlier in this branch... I'm seeing segfaults in the regular leapfrog test at cycle 27 again. Trying to debug now, although gdb says it's not seeing any debug symbols even when I built with `RelWithDebInfo`

Reply:
quick update on the particles failure: I'm getting a segfault on this line:
```c++
      PARTHENON_MPI_CHECK(MPI_Isend(bd_var_.send[n].data(), send_size[n],                            
                                    MPI_PARTHENON_REAL, nb.snb.rank, send_tag[n],                    
                                    MPI_COMM_WORLD, &(bd_var_.req_send[nb.bufid])));
```
(line ~100 in `src/bvals/bvals_swarm.cpp`; I checked the macro expansion and it is the MPI call that fails when I run two mpi processes with xterm and gdb on a power9 node with cuda backend). Not clear to me what the segfault is since I can print out `bd_var_.send[n].extent(0)`, `send_size[n]`, `send_tag[n]`, and I can write to `bd_var_.req_send[nb.bufid]` and `nb.snb.rank=1` (the error is occuring on rank 0). I'll keep looking, just a bit confused right now.

Reply:
The error: 
![image](https://user-images.githubusercontent.com/2207688/133131554-f8492b89-59c6-4428-a6b0-53ab4c13cbe7.png)